### PR TITLE
net-mgmt/net-snmp: Add support for agentx

### PR DESCRIPTION
**Pull request by joshland**
- Add boolean variable for AgentX support to model
 - Add to checkbox to General form.
 - Add agentx config to template.
 - Answers https://github.com/opnsense/plugins/issues/1200
(split out from multi-plugin patch previously submitted)

### CODE REVIEW

File: net-mgmt/net-snmp/src/opnsense/mvc/app/models/OPNsense/Netsnmp/General.xml [hunk 1]
```diff
@@ -27,6 +27,10 @@
             <default>0</default>
             <Required>Y</Required>
         </versionoid>
+        <enableagentx type="BooleanField">
+            <default>0</default>
+            <Required>Y</Required>
```
Comment (fichtner) [inline net-mgmt/net-snmp/src/opnsense/mvc/app/models/OPNsense/Netsnmp/General.xml:32]: Model bump required at the top of this file

File: net-mgmt/net-snmp/src/opnsense/service/templates/OPNsense/Netsnmp/snmpd.conf [hunk 2]
```diff
@@ -12,6 +12,11 @@ agentAddress udp6:[{{ network }}]:161
 agentAddress udp:161,udp6:[::1]:161
 {% endif %}
 
+{% if helpers.exists('OPNsense.netsnmp.general.enableagentx') and OPNsense.netsnmp.general.enableagentx == '1' %}
```
Comment (fichtner) [inline net-mgmt/net-snmp/src/opnsense/service/templates/OPNsense/Netsnmp/snmpd.conf:15]: If we force required and migrate model we can omit the exists check.

**Review by mimugmail — APPROVED** [review]
Looks good,  thank you!

- Please version bump in Makefile
- Please add changelog tp pkg-descr
- Things @fichtner metioned

**Comment by joshland** [comment]
- resync with upstream/master
 - bump version in Makefile
 - add changelog
 - Uncertain about forcing Migation/etc.

File: net-mgmt/net-snmp/src/opnsense/service/templates/OPNsense/Netsnmp/snmpd.conf (hunk 2 above)
Comment (fichtner) [inline net-mgmt/net-snmp/src/opnsense/service/templates/OPNsense/Netsnmp/snmpd.conf:15]:
```suggestion
{% if OPNsense.netsnmp.general.enableagentx == '1' %}
```

**Review by fichtner — APPROVED** [review]

**Comment by fichtner** [comment]
Merged, thanks!

**Pull request merged**